R/class-yarn.R

```diff
@@ -84,15 +84,18 @@ yarn <- R6::R6Class("yarn",
     #' ex1$write(tmp)
     #' head(readLines(tmp)) # now a markdown file
     #' unlink(tmp)
-    write = function(path = NULL, 
+    write = function(path = NULL,
       stylesheet_path = system.file("extdata", "xml2md_gfm.xsl", package = "tinkr")){
+
+      output <- to_md(self, path, stylesheet_path)
+
       if (is.null(path)) {
-        stop("Please provide a file path", call. = FALSE)
+        cat(output)
       }
-      to_md(self, path, stylesheet_path)
-      invisible(self)
```

Comment:
This part is important for chaining R6 methods (described in the last paragraph of the "Private Members" section: https://r6.r-lib.org/articles/Introduction.html#private-members)